.github/workflows/rust.yml

```diff
@@ -0,0 +1,92 @@
+name: Rust
+
+on:
+  workflow_dispatch:
+  push:
+    branches: [ main ]
+    paths:
+      - 'sdk/rust/**'
+      - '!sdk/rust/README.md'
+      - 'lib/beaglebone/**'
+      - 'lib/linux/**'
+      - 'lib/jetson/**'
+      - 'lib/mac/**'
+      - 'lib/raspberry-pi/**'
+      - 'lib/windows/**'
+      - '.github/workflows/rust.yml'
+  pull_request:
+    branches: [ main ]
+    paths:
+      - 'sdk/rust/**'
+      - '!sdk/rust/README.md'
+      - 'lib/beaglebone/**'
+      - 'lib/linux/**'
+      - 'lib/jetson/**'
+      - 'lib/mac/**'
+      - 'lib/raspberry-pi/**'
+      - 'lib/windows/**'
+      - '.github/workflows/rust.yml'
+
+env:
+  SDL_AUDIODRIVER: "disk"
+
+defaults:
+  run:
+    working-directory: sdk/rust
+    shell: bash
+
+jobs:
+  build-github-hosted:
+    runs-on: ${{ matrix.os }}
+
+    strategy:
+      matrix:
+        os: [ubuntu-latest, windows-latest, macos-latest]
+
+    steps:
+    - uses: actions/checkout@v2
+      with:
+        submodules: recursive
+
+    - name: Rust pre-build
+      run: bash copy.sh
+
+    - name: Install stable toolchain
+      uses: actions-rs/toolchain@v1
+      with:
+        profile: minimal
+        toolchain: stable
+        override: true
+
+    - name: Build
+      run: cargo build --verbose
+
+    - name: Run get audio devices
+      run: cargo run --release --example demo -- --show_audio_devices
+
+  build-self-hosted:
+    runs-on: ${{ matrix.machine }}
+
+    strategy:
+      matrix:
+        machine: [rpi2, rpi3-32, rpi3-64, rpi4-32, rpi4-64, jetson, beaglebone]
+
+    steps:
+    - uses: actions/checkout@v2
+      with:
+        submodules: recursive
+
+    - name: Rust pre-build
+      run: bash copy.sh
+
+    - name: Install stable toolchain
+      uses: actions-rs/toolchain@v1
+      with:
+        toolchain: nightly
```

Comment:
Do we need nightly? Probably best to test with a stable version because if we can't build on stable that's an issue

Reply:
Oh, okay. I suppose one way to look at is that we use stable on the github hosted ones, so actually this increases coverage because it's an additional test with nightly.

Reply:
yes github ones use stable

Reply:
Okay, in that case let's just leave it as is to keep the pattern - Sorry for the trouble